messaging/types/persistence.go

```diff
@@ -1,9 +1,13 @@
 package types
 
-import "crypto/ecdsa"
+import (
+	"crypto/ecdsa"
+)
 
 type Persistence interface {
-	wakuPersistence
+	WakuStorage() WakuPersistence
+	SegmentationStorage() SegmentationPersistence
+	EncryptionStorage() EncryptionPersistence
```

Comment:
Would it not be simpler to compose instead?

```suggestion
type Persistence interface {
	WakuPersistence
	SegmentationPersistence
	EncryptionPersistence
```

In case of `status-go` it is going to be a single implementation, right?
And in case they're separate, one could still compose all implementations into one

Reply:
I prefer the accessor-based version for several reasons:

- it avoids name collisions. For example, if both `Waku` and `Transport` expose an `AddKeys` method, composition would lead to conflicts
- it provides an explicit namespace for each sub-component. Calls like `persistence.Waku().AddKeys()` are clearer at the call site, as it is immediately obvious which scope the method belongs to
- It simplifies mocking. If only one sub-component of persistence needs to be mocked, it can return nil for the rest. With composition flat-model, all other methods would also have to be implemented, even if unused